src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -7617,11 +7618,11 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* putArgStk)
     {
         if (data->IsIconHandle())
         {
-            inst_IV_handle(INS_push, data->AsIntCon()->gtIconVal);
+            inst_IV_handle(INS_push, (target_ssize_t)data->AsIntCon()->gtIconVal);
```

Comment:
I don't think this is correct. For x64 hosted targeting x86, I presume gtIconVal will be a 64-bit pointer than needs to be relocated, but this will truncated it to 32 bits.